src/components/widgets/AltDateWidget.js

```diff
@@ -100,10 +103,15 @@ class AltDateWidget extends Component {
   };
 
   get dateElementProps() {
-    const { time } = this.props;
+    const { time, options } = this.props;
     const { year, month, day, hour, minute, second } = this.state;
     const data = [
-      { type: "year", range: [1900, 2020], value: year },
+      {
+        type: "year",
+        range:
+          options.yearsRange || AltDateWidget.defaultProps.options.yearsRange,
```

Comment:
I don't like this change that much, however only using `options.yearsRange` is not working while documentation says:

> If you need to pass options to your custom widget, you can add a ui:options object containing those properties. If the widget has defaultProps, the options will be merged with the (optional) options object from defaultProps

This merge is not taking place here. If I don't provide any `yearsRange` to uiSchema options object only contains an `enumOptions` prop.

Reply:
Are you sure? There are a few tests that cover that the `ui:options` merge happens -- see uiSchema_test.js -- and when I take this code out, it seems to fall back safely to `[1900, 2020]` (i.e. this year plus 2).

Reply:
@glasserc I saw this test too, but trying to removing the `||` code break the demo app.

I investigated this a little more, seems that the issue is only with the _AltDateTimeWidget_ but not with _AltDateWidget_. That's weird as AltDateTimeWidget is only a very simple wrapper as far I can see.

Recap: if I try to rely on default values I have:

* _AltDateWidget_ with options `{yearsRange: Array(2), enumOptions: false}`
* _AltDateTimeWidget_ with options `{enumOptions: false}`

(I rebased on master to the sure of that, nothing changed).

Reply:
This is probably because of `AltDateTimeWidget` being a wrapper around `AltDateWidget` that doesn't have its own `defaultProps` set. I think the right thing to do here is to set `AltDateTimeWidget.defaultProps = AltDateWidget.defaultProps`.

Reply:
@glasserc uh! That was easy, sorry for not spotted this before.
Yes, it works properly now, PR updated!